From 7267cb85732adc3d879808221b96f48979fea161 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:46 -0400
Subject: [tools] lib/traceevent: Unregister handler when xen plugin is unloaded

Message-id: <1408953210-26343-279-git-send-email-jolsa@redhat.com>
Patchwork-id: 89029
O-Subject: [PATCH RHEL7.1 BZ1133083 278/322] tools lib traceevent: Unregister handler when xen plugin is unloaded
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit bf6b3a95ff439b1dcd6151b3f38810f3cec1e319
Author: Namhyung Kim <namhyung@kernel.org>
Date: Thu Jan 16 11:31:18 2014 +0900

description
===========
The function handler should be unregistered when the plugin is unloaded
otherwise it'll try to access invalid memory.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/plugin_xen.c b/tools/lib/traceevent/plugin_xen.c
index e779429..3a413ea 100644
--- a/tools/lib/traceevent/plugin_xen.c
+++ b/tools/lib/traceevent/plugin_xen.c
@@ -128,3 +128,9 @@ int PEVENT_PLUGIN_LOADER(struct pevent *pevent)
            PEVENT_FUNC_ARG_VOID);
  return 0;
 }
+
+void PEVENT_PLUGIN_UNLOADER(struct pevent *pevent)
+{
+ pevent_unregister_print_function(pevent, process_xen_hypercall_name,
+      "xen_hypercall_name");
+}
-- 
1.7.1